pyro/distributions/util.py

```diff
@@ -5,6 +5,28 @@
 from torch.autograd import Variable
 
 
+def broadcast_shape(*shapes):
+    """
+    Similar to ``np.broadcast()`` but for shapes.
+    Equivalent to ``np.broadcast(*map(np.empty, shapes)).shape``.
+
+    :param tuple shapes: shapes of tensors.
+    :returns: broadcasted shape
+    :rtype: tuple
+    :raises: ValueError
+    """
+    reversed_shape = []
+    for shape in shapes:
+        for i, size in enumerate(reversed(shape)):
+            if i >= len(reversed_shape):
+                reversed_shape.append(size)
+            elif reversed_shape[i] == 1:
+                reversed_shape[i] = size
+            elif size != 1 and reversed_shape[i] != size:
+                raise ValueError('shape mismatch: objects cannot be broadcast to a single shape')
```

Comment:
Do you think it will be useful to have the tensor sizes in the error statement when broadcasting fails?

Reply:
Maybe. This version prints the `np.broadcast()` error message verbatim. I can change it if you feel strongly.

Reply:
Lets leave it as is for now. We can put in a more verbose message if we need it later.